src/js/components/SelectMultiple/__tests__/SelectMultiple-test.tsx

```diff
@@ -455,5 +457,5 @@ describe('SelectMultiple', () => {
     expect(
       screen.queryByRole('option', { name: /Strawberry selected/ }),
     ).toBeNull();
-  });
+  }, 10000);
```

Comment:
Um! From what I understand the test 'search with select and clear' has some issues with it. It is working fine mostly but sometimes it just takes a little longer than its average time.

Reply:
Yeah it feels like maybe we should split up the test into multiple tests so that we don't have 1 long running test. From what I have noticed other tests seem to timeout only if this select multiple test first times out. It seems like there is a cascade effect. I think it we break up the test into smaller chunks this will improve readability and solve our timeout issue.

Reply:
How should I split up the test? Should I create another file that contains half of the tests?

Reply:
![image](https://github.com/grommet/grommet/assets/98217730/ca3dd557-6cfc-457d-a2a3-a184db593cc3)
After testing SelectMultiple individually it seems like this one test 'search with select and clear' is causing all the problems

Reply:
Update I tried to optimize the test a little here's the code

```   
test('search with select and clear', async () => {
    const user = userEvent.setup();


const defaultOptions = [
      'Apple',
      'Orange',
      'Banana',
      'Grape',
      'Melon',
      'Strawberry',
      'Kiwi',
      'Mango',
      'Raspberry',
      'Rhubarb',
    ];

    const Test = () => {
      const [options, setOptions] = useState(defaultOptions);
      const [valueMultiple, setValueMultiple] = useState([]);

      const handleSearch = (text: string) => {
        const filtered = defaultOptions.filter((option) =>
          option.toLowerCase().includes(text.toLowerCase()),
        );
        setOptions(filtered);
      };

      return (
        <Grommet>
          <SelectMultiple
            options={options}
            value={valueMultiple}
            placeholder="Select"
            onSearch={handleSearch}
            onClose={() => setOptions(defaultOptions)}
            onChange={({ value }) => {
              setValueMultiple(value);
            }}
          />
        </Grommet>
      );
    };

    render(<Test />);
    // open drop
    await user.click(screen.getByRole('button', { name: /Select/ }));
    // search
    await user.type(screen.getByRole('searchbox', { name: /Search/ }), 'p');
    // select all
    await user.click(screen.getByRole('button', { name: /Select all/ }));

    const searchInput = screen.getByRole('searchbox', { name: /Search/ });
    const optionSelectedInput = screen.queryAllByRole('option', {
      name: /selected$/,
    });

    expect(optionSelectedInput).not.toBeNull();

    // search
    await user.type(searchInput, '{backspace}w');
    // select all
    await user.click(screen.getByRole('button', { name: /Select all/ }));

    expect(optionSelectedInput).not.toBeNull();

    // search
    await user.type(searchInput, '{backspace}b');
    // Clear
    await user.click(screen.getByRole('button', { name: /Clear all/ }));
    // clear search
    await user.type(searchInput, '{backspace}');
    expect(
      screen.queryByRole('option', { name: /Grape selected/ }),
    ).not.toBeNull();
    expect(
      screen.queryByRole('option', { name: /Strawberry selected/ }),
    ).toBeNull();
  });
```
  Still, the test takes 4869 ms on average and times out when tested with other tests.
  I tried to create a separate file for this test as well which also resulted in the same timeout issue.
  I guess setting a timeout value is the only solution left.